cloudferrylib/os/actions/live_migration.py

```diff
@@ -0,0 +1,264 @@
+
+            destroyer = libvirt.DestNovaInstanceDestroyer(dst_libvirt,
+                                                          dst_compute,
+                                                          dst_virsh_name,
+                                                          dst_instance.id)
+            try:
+                destroyer.do()
+                src_libvirt.live_migrate(src_virsh_name, dst_host, migration_file.filename)
+            except remote_runner.RemoteExecutionError:
+                destroyer.undo()
+            finally:
+                dst_libvirt.move_backing_file(dst_backing_file, new_id)
+
+
+class UpdateInstancesAutoIncrement(action.Action):
+    """Updates `nova.instances` auto increment value on destination. This is
+    required to match libvirt instance IDs on source and destination. Libvirt
+    instance IDs must be identical on source and destination for live migration
+
+    Requirements:
+      - Write access to nova DB on destination
+    """
+    def run(self, info=None, **kwargs):
+        instance_id = info[utils.INSTANCES_TYPE].keys()[0]
+        mysql_id = self.src_cloud.resources[utils.COMPUTE_RESOURCE].\
+            get_instance_sql_id_by_uuid(instance_id)
+        self.dst_cloud.resources[
+            utils.COMPUTE_RESOURCE].update_instance_auto_increment(mysql_id)
+
+
+class RestoreOriginalInstancesTableSchema(action.Action):
+    """Restores original value of `nova.instances` auto-increment. Should be
+    used together with `UpdateInstancesAutoIncrement`
+
+    Requirements:
+      - Write access to nova DB on destination
+    """
+    def run(self, info=None, **kwargs):
+        nova_compute = self.dst_cloud.resources[utils.COMPUTE_RESOURCE]
+        mysql = nova_compute.mysql_connector
+        reset_instances_id = ("alter table instances change id id INT (11) "
+                              "not null auto_increment")
+        LOG.info("Resetting nova.instances.id column to original value")
+        mysql.execute(reset_instances_id)
+
+
+class CheckQemuVersion(action.Action):
+    """Verifies qemu version is 2.0 on all source and destination compute
+    nodes. This is the primary requirement for live migration.
+
+    Requirements:
+     - ssh access to compute nodes
+    """
+
+    REQUIRED_QEMU_VERSION = "2.0"
+
+    def run(self, info=None, **kwargs):
+        faulty_hosts = []
+
+        for cloud in [self.src_cloud, self.dst_cloud]:
+            nova_compute = cloud.resources[utils.COMPUTE_RESOURCE]
+            hosts = nova_compute.get_hypervisors()
+
+            for host in hosts:
+                runner = remote_runner.RemoteRunner(
+                    host, cloud.cloud_config.cloud.ssh_user)
+
+                qemu_version_cmd = ("kvm -version  | "
+                                    "sed -E 's/QEMU emulator version "
+                                    "([0-9]\.[0-9].?[0-9]?).*/\\1/'")
+
+                version = runner.run(qemu_version_cmd)
```

Comment:
I've got an error here. `NetworkError: Name lookup failed for grizzly`
Because host was taken as `hosts = nova_compute.get_hypervisors()` it means than CF will work with hostnames only which not resolvable from CF node.